pkg/webhooks/regularuser/common/regularuser.go

```diff
@@ -211,36 +216,55 @@ func (s *RegularuserWebhook) authorized(request admissionctl.Request) admissionc
 		ret.UID = request.AdmissionRequest.UID
 		return ret
 	}
-	if strings.HasPrefix(request.AdmissionRequest.UserInfo.Username, "system:") {
-		ret = admissionctl.Allowed("authenticated system: users are allowed")
-		ret.UID = request.AdmissionRequest.UID
-		return ret
-	}
+
 	if strings.HasPrefix(request.AdmissionRequest.UserInfo.Username, "kube:") {
 		ret = admissionctl.Allowed("kube: users are allowed")
 		ret.UID = request.AdmissionRequest.UID
 		return ret
 	}
-	if isCustomDomainAuthorized(request) {
-		ret = admissionctl.Allowed("Management of CustomDomain CR is authorized")
-		ret.UID = request.AdmissionRequest.UID
-		return ret
-	}
-	if isMustGatherAuthorized(request) {
-		ret = admissionctl.Allowed("Management of MustGather CR is authorized")
-		ret.UID = request.AdmissionRequest.UID
-		return ret
+
+	switch {
+	case request.Kind.Kind == mustGatherKind && request.Kind.Group == mustGatherGroup:
```

Comment:
Could we encapsulate this in a function to improve readability?
```suggestion
	case isMustGatherReq(request):
```

Reply:
I'm down, what do you think about something like:

```go
func requestMatchesGroupKind(req admissionctl.Request, kind, group string) bool
```

So we don't have to make a bunch of `is${KIND}Req` functions?